miio/integrations/zhimi/humidifier/airhumidifier_miot.py

```diff
@@ -381,3 +540,238 @@ def set_dry(self, dry: bool):
+            {'did': 'child_lock', 'siid': 6, 'piid': 1, 'code': 0, 'value': False},
+            {'did': 'actual_speed', 'siid': 7, 'piid': 1, 'code': 0, 'value': 1100},
+            {'did': 'clean_mode', 'siid': 7, 'piid': 5, 'code': 0, 'value': False}
+            {'did': 'self_clean_percent, 'siid': 7, 'piid': 6, 'code': 0, 'value': 0},
+            {'did': 'pump_state, 'siid': 7, 'piid': 7, 'code': 0, 'value': False},
+            {'did': 'pump_cnt', 'siid': 7, 'piid': 8, 'code': 0, 'value': 1000},
+        ]
+    """
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+        super().__init__(self.data)
+        self.embed("common", AirHumidifierMiotCommonStatus(self.data))
+
+    # Air Humidifier 3
+
+    @property
+    @setting(
+        name="Operation Mode",
+        setter_name="set_mode",
+    )
+    def mode(self) -> OperationModeCA6:
+        """Return current operation mode."""
+
+        try:
+            mode = OperationModeCA6(self.data["mode"])
+        except ValueError as e:
+            _LOGGER.exception("Cannot parse mode: %s", e)
+            return OperationModeCA6.Auto
+
+        return mode
+
+    @property
+    @sensor(
+        "Water Level",
+        unit="%",
+        device_class="measurement",
+        icon="mdi:water-check",
+        entity_category="diagnostic",
+    )
+    def water_level(self) -> Optional[int]:
+        """Return current water level (empty/min, normal, full/max).
+
+        0 - empty/min,  1 - normal, 2 - full/max
+        """
+        water_level = self.data["water_level"]
+        return {0: 0, 1: 50, 2: 100}.get(water_level)
+
+    @property
+    @sensor(
+        "Operation status",
+        device_class="measurement",
+        entity_category="diagnostic",
+    )
+    def status(self) -> OperationStatusCA6:
+        """Return current status."""
+
+        try:
+            status = OperationStatusCA6(self.data["status"])
+        except ValueError as e:
+            _LOGGER.exception("Cannot parse status: %s", e)
+            return OperationStatusCA6.Close
+
+        return status
+
+    # Other
+
+    @property
+    @sensor(
+        "Self-clean Percent",
+        unit="s",
+        device_class="total_increasing",
+        icon="mdi:progress-clock",
+        entity_category="diagnostic",
+    )
+    def self_clean_percent(self) -> int:
+        """Return time in minutes (from 0 to 30) of self-cleaning procedure."""
+        return self.data["self_clean_percent"]
+
+    @property
+    @sensor(
+        "Pump State",
+        entity_category="diagnostic",
+    )
+    def pump_state(self) -> bool:
+        """Return pump state."""
+        return self.data["pump_state"]
+
+    @property
+    @sensor(
+        "Pump Cnt",
+        entity_category="diagnostic",
+    )
+    def pump_cnt(self) -> int:
+        """Return pump-cnt."""
+        return self.data["pump_cnt"]
+
+
+class AirHumidifierMiotCA6(MiotDevice):
```

Comment:
Are the differences so large between the models, that the implementation for CA6 couldn't fit in the same implementation class with if-guarding based on the model?

Reply:
I think, that the differences are big. CA6 introduced new properties, like `pump_cnt`, and changed behaviour of some other props. I already introduced some common thing in classes like `AirHumidifierMiotCommonStatus`. Moreover, I'm pretty sure that the manufacturer will not stop added/changing things in new models. Thus, it's better to be prepared. If-guarding in this case would not be convenient.